### ISSUE TITLE
1.3.0 breaks git submodules

### ISSUE THREAD

**Issue by roschaefer** [rwjblue/git-repo-info#27]
I have a git repository containing my ember frontend and rails backend as git submodules. I run full-stack tests within the parent repository. Tonight my travis build on a feature branch was suddenly broken and even master after a rebuild: 
[ENOENT: no such file or directory, open '../.git/modules/frontend/commondir'](https://travis-ci.org/roschaefer/rundfunk-mitbestimmen/builds/168463624)

Downgrading to `git-repo-info@1.2.0` with `npm shrinkwrap` and updating the parent repository to the new commit fixed the issue.

**Comment by kirrg001** [comment]
+1

**Comment by rwjblue** [comment]
Can someone push up a set of repos that demonstrate the issue? I'm happy to dig in and try to fix once I understand what is happening...

**Comment by ISaidHey** [comment]
At the moment I can't upload a submodule set (will do tonight if no one else can) but I can confirm that this is breaking my companies repo and preventing deploy. 

```
ENOENT: no such file or directory, open '../.git/modules/sh-app/commondir'
Error: ENOENT: no such file or directory, open '../.git/modules/sh-app/commondir'
    at Error (native)
    at Object.fs.openSync (fs.js:584:18)
    at Object.fs.readFileSync (fs.js:431:33)
    at findRepoHandleLinkedWorktree (/data/sh-app/node_modules/git-repo-info/index.js:26:32)
    at findRepo (/data/sh-app/node_modules/git-repo-info/index.js:46:14)
    at module.exports (/data/sh-app/node_modules/git-repo-info/index.js:144:21)
    at version (/data/sh-app/node_modules/git-repo-version/index.js:11:14)
    at CoreObject.module.exports.config (/data/sh-app/node_modules/ember-cli-app-version/index.js:10:46)
    at CoreObject.superWrapper [as config] (/data/sh-app/node_modules/core-object/lib/assign-properties.js:32:18)
    at /data/sh-app/node_modules/ember-cli/lib/models/project.js:226:27
```

And when I go and check, there is in fact no file/folder

**Comment by roschaefer** [comment]
If you need an example, you can take my repository: https://github.com/roschaefer/rundfunk-mitbestimmen
I set the submodule repository ("frontend") with the downgraded version at [8ca70a6](https://github.com/roschaefer/rundfunk-mitbestimmen/commit/8ca70a6a0f8b96077f61c676c39c1e33d55594ba)

Any commit below that should reproduce the issue, e.g. [9f74982](https://github.com/roschaefer/rundfunk-mitbestimmen/commit/9f7498276e123fc0cd26c431806bea47ba61d0f4)

And @ISaidHey is correct, there is no file or folder `commondir`. I guess, `git-repo-info` shouldn't look for that?

EDIT: I should mention that I didn't try to reproduce the error on my machine. When I got the email from Travis CI about the broken build I got very irritated and immediately expected a dependency issue. Maybe the Travis build logs will help you to track down the problem.

**Comment by kevinansfield** [comment]
repro steps with Ghost-Admin but can be any ember app:

```
mkdir submodule-test
cd submodule-test
git init
git submodule add git@github.com:TryGhost/Ghost-Admin.git
cd Ghost-Admin
npm install && bower install
```

From that point on any ember build tasks run inside the Ghost-Admin directory will fail.

**Comment by ErisDS** [comment]
Just reaching out to say that this is quite heavily impacting us @ Team Ghost. It's hard to get a fresh build working atm, and therefore really really hard for us to build releases.

I also flagged this upstream: https://github.com/cibernox/git-repo-version/issues/14

**Comment by hjdivad** [comment]
@rwjblue ok from what i can tell submodules were never actually supported, but submodules did not cause an exception to throw.  I'll get a PR up to resolve the regression but defer proper submodule support.

**Comment by hjdivad** [comment]
Thanks for your repro steps.  This should be fixed by #29 for modern git submodule layouts.

**Comment by hjdivad** [comment]
Please feel free to ping me if you discover that not to be the case.
